apps/remix-ide/src/app/files/dgitProvider.js

```diff
@@ -0,0 +1,398 @@
+'use strict'
+
+import {
+  Plugin
+} from '@remixproject/engine'
+import git from 'isomorphic-git'
+import IpfsHttpClient from 'ipfs-http-client'
+import {
+  saveAs
+} from 'file-saver'
+
+const JSZip = require('jszip')
+const path = require('path')
+const FormData = require('form-data')
+const axios = require('axios')
+
+const profile = {
+  name: 'dGitProvider',
+  displayName: 'Decentralized git',
+  description: '',
+  icon: 'assets/img/fileManager.webp',
+  version: '0.0.1',
+  methods: ['init', 'status', 'log', 'commit', 'add', 'remove', 'rm', 'lsfiles', 'readblob', 'resolveref', 'branches', 'branch', 'checkout', 'currentbranch', 'push', 'pin', 'pull', 'pinList', 'unPin', 'setIpfsConfig', 'zip', 'setItem', 'getItem'],
+  kind: 'file-system'
+}
+class DGitProvider extends Plugin {
+  constructor (fileManager) {
+    super(profile)
+    this.fileManager = fileManager
+    this.ipfsconfig = {
+      host: 'ipfs.komputing.org',
+      port: 443,
+      protocol: 'https',
+      ipfsurl: 'https://ipfsgw.komputing.org/ipfs/'
+    }
+    this.globalIPFSConfig = {
+      host: 'ipfs.io',
+      port: 443,
+      protocol: 'https',
+      ipfsurl: 'https://ipfs.io/ipfs/'
+    }
+  }
+
+  async getGitConfig () {
+    const workspace = await this.call('filePanel', 'getCurrentWorkspace')
+    return {
+      fs: window.remixFileSystem,
+      dir: `.workspaces/${workspace.name}`
```

Comment:
we could do without hardcoding the value. see https://github.com/ethereum/remix-project/blob/master/apps/remix-ide/src/app/files/workspaceFileProvider.js#L10

Reply:
ok I handled it.